### DISCUSSION TITLE
Sumatra tabs

### DISCUSSION THREAD

**Discussion by salavata** [sumatrapdfreader/sumatrapdf#4310] [General]
Hi!
A few days ago I opened an issue:
https://github.com/sumatrapdfreader/sumatrapdf/issues/4304
In fact, it contained three closely related issues in one:
1) Changing the TabWidth value does not really change the maximum width of tabs.
2) The "left-right" arrow buttons overlapped the cross and prevented a user from closing the tab.
3) Tabs do not widen or shrink depending on the filenames, so tabs of books with similar starting names (e.g. "Crystallization" and "Crystallography") look identical.

The issue was closed because the 3rd is too complex. ("Tabs dont alter due to text that would be too complex").
However, the 1st and the 2nd are remaining!
And the 1st and the 2nd are not complex, because the 1st bug was not present 3.4.6.

Is it possible to revive the issue?

**Comment by kjk** [comment]
`TabWidth` has just been fixed. 

The second one is just as hard as the 3rd one: those are displayed by Windows's lousy tab control. To fix them I would have to re-implement the whole tab control.

**Comment by l-33ter** [comment]
Hi all,

I believe my request will be successful here. As my picture indicates, on a 4K screen, tabs are incredibly small. However, from what I understand of @kjk's post, this issue will coincide with the poor tab control in Windows, correct?

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/77815858/4a4df009-72fc-47d2-a66d-4af8dcce0dc7)

Would it be possible to give the tab viewer an additional line?

**Reply by GitHubRulesOK** [reply]
Hmmm the core concept is minimal UI maximal viewing hence few icons and 
originally NO tabs much like core Spartan by powerful MuPDF which has no tabs and no icons I.E. keyboard or mouse only the odd pop up areas

You can alter the scale of tabs via UI font size and tab length but how many concurrent pages are needed at one time?
(personally rarely need more than 2-4 one for read /write and one or a few for cross reference.)
more than a few seems inefficient use of cpu etc. why not 2 or more tabbed sessions each with its own desktop for full screen view ?
